instrument-starters/opentracing-spring-cloud-core/src/main/java/io/opentracing/contrib/spring/cloud/async/ExecutorBeanPostProcessor.java

```diff
@@ -52,49 +80,72 @@ public Object postProcessBeforeInitialization(Object bean, String beanName)
 
   @Override
   public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
-    if (bean instanceof Executor && !(bean instanceof ThreadPoolTaskExecutor)) {
-      Method execute = ReflectionUtils.findMethod(bean.getClass(), "execute", Runnable.class);
-      boolean methodFinal = Modifier.isFinal(execute.getModifiers());
-      boolean classFinal = Modifier.isFinal(bean.getClass().getModifiers());
-      boolean cglibProxy = !methodFinal && !classFinal;
-      Executor executor = (Executor) bean;
-      ProxyFactoryBean factory = new ProxyFactoryBean();
-      factory.setProxyTargetClass(cglibProxy);
-      factory.addAdvice(new ExecutorMethodInterceptor<>(executor, tracer));
-      factory.setTarget(bean);
-      return factory.getObject();
-    } else if (bean instanceof ThreadPoolTaskExecutor) {
-      boolean classFinal = Modifier.isFinal(bean.getClass().getModifiers());
-      boolean cglibProxy = !classFinal;
-      ThreadPoolTaskExecutor executor = (ThreadPoolTaskExecutor) bean;
-      ProxyFactoryBean factory = new ProxyFactoryBean();
-      factory.setProxyTargetClass(cglibProxy);
-      factory.addAdvice(new ExecutorMethodInterceptor<ThreadPoolTaskExecutor>(executor, tracer) {
-        @Override
-        Executor tracedExecutor(Tracer tracer, ThreadPoolTaskExecutor executor) {
-          return new TracedThreadPoolTaskExecutor(tracer, executor);
-        }
-      });
-      factory.setTarget(bean);
-      return factory.getObject();
+    if (bean instanceof Executor) {
+      if (bean instanceof ExecutorService) {
+        ExecutorService executorService = (ExecutorService) bean;
+        return proxify(
+            executorService,
+            TracedExecutorService::new,
+            shouldUseCGLibProxy(executorService, TRACED_EXECUTOR_SERVICE_METHODS)
+        );
+      } else if (bean instanceof ThreadPoolTaskExecutor) {
+        ThreadPoolTaskExecutor threadPoolTaskExecutor = (ThreadPoolTaskExecutor) bean;
+        return proxify(
+            threadPoolTaskExecutor,
+            (e, t) -> new TracedThreadPoolTaskExecutor(t, e),
+            shouldUseCGLibProxy(threadPoolTaskExecutor, TRACED_THREAD_POOL_TASK_EXECUTOR_METHODS)
+        );
+      } else {
```

Comment:
@ZawadaZSE how do you feel about adding another block after `ThreadPoolTaskExecutor` for `AsyncTaskExecutor`? It would give proper tracing and cglib proxy for classes such as `ThreadPoolTaskScheduler` . Issue #228 would be fixed by that 

I am also ok adding that in after your PR goes in. 

Thanks

Reply:
@ZawadaZSE what are your thoughts on the above request? Thanks.